### ISSUE TITLE
Branching strategy [was: Improvements to the release tools]

### ISSUE THREAD

**Issue by razvan** [stackabletech/stackable-utils#5]
Related articles:
 * Branch strategies: https://www.gitkraken.com/learn/git/best-practices/git-branch-strategy

Open questions to the release process: 

- [ ] The lack of a development branch together with the `cargo deny `restriction that only tags can be referenced by dependecies means that for every feature in `operator-rs`, a release is required.
- [ ] Bugfix-Releases cannot be done currently if we only release from `main`. They require release branches (to host the tags) and  or the possibility to cherry-pick commits to main,
- [ ] Have release candidate tags in release branches so that upstream can test them. This is especially needed for operators that cannot depend on `operator-rs` branches anymore.
- [x]  Tags/Releases - are they protected from deletion? No! Apparently not possible [1]. **Decision**: we'll have to live with it for now.

Possible changes to the release tools:

- [ ] After a release: increment the patch level instead of the minor level.
- [ ] `git pull` in main before starting a release. Is this really a good idea ?
- [ ] Check if release tag already exists and stop if it does.
- [ ] Switch back to the original branch when done
- [ ] Delete the local release branch when done.
- [x] Update the change log file.

[1] https://stackoverflow.com/questions/45791080/restrict-users-from-deleting-tag-in-github.

**Comment by lfrancke** [comment]
@razvan Is this basically done with the discussion you had today and the followup ticket you created?
